lib/static/new-ui/features/suites/components/SuitesTreeView/index.tsx

```diff
@@ -59,39 +60,56 @@ function SuitesTreeViewInternal(props: SuitesTreeViewProps): ReactNode {
     const virtualizer = useVirtualizer({
         count: list.structure.visibleFlattenIds.length,
         getScrollElement: () => parentRef.current,
-        estimateSize: () => 28,
+        estimateSize: () => 32,
         getItemKey: useCallback((index: number) => list.structure.visibleFlattenIds[index], [list]),
-        overscan: 200
+        overscan: 50
     });
 
     const virtualizedItems = virtualizer.getVirtualItems();
 
+    useImperativeHandle(ref, () => ({
+        scrollToId: (id: string): void => {
+            virtualizer.scrollToIndex(list.structure.visibleFlattenIds.indexOf(id), {align: 'center'});
+        }
+    }));
+
     // Effects
     useEffect(() => {
-        if (!props.isInitialized) {
+        if (!isInitialized) {
             return;
         }
 
-        props.actions.setStrictMatchFilter(false);
+        dispatch(setStrictMatchFilter(false));
 
+        let timeoutId: NodeJS.Timeout;
         if (suiteId) {
-            props.actions.suitesPageSetCurrentSuite(suiteId);
-            virtualizer.scrollToIndex(list.structure.visibleFlattenIds.indexOf(suiteId), {align: 'start'});
+            dispatch(suitesPageSetCurrentSuite(suiteId));
+            timeoutId = setTimeout(() => {
```

Comment:
This tiny timeout is needed to postpone scrolling to current item. Otherwise, when loading thousands of items, sometimes scroll position can be incorrect.